src/zenml/analytics/utils.py

```diff
@@ -101,11 +102,13 @@ class analytics_disabler:
 
     def __init__(self) -> None:
         """Initialization of the context manager."""
-        self.original_value = os.environ.get(ENV_ZENML_ANALYTICS_OPT_IN)
+        self.original_value: Optional[bool] = None
 
     def __enter__(self) -> None:
         """Disable the analytics."""
-        os.environ[ENV_ZENML_ANALYTICS_OPT_IN] = str(False)
+        global _INTERNAL_DISABLE_ANALYTICS
```

Comment:
@bcdurak I had to make some changes to this analytics disable context manager, because it lead to issues with the default stack registration. The following was happening:
- The `SQLZenStore` was initialized, was caused the flavors to be registered.
- The flavor registration caused an analytics call
- The analytics call initialized the zen store of the global config
- The zen store initialization tried to register the default stack
- `orchestrator_schema.to_model()` failed because the flavor was not registered yet

I broke this loop by disabling the analytics before it even gets to the global config. Does this make sense or do I mess stuff up in other places by doing this?

Reply:
This is a crazy connection but I can not think of any problems that this change would cause 👍